### PR TITLE
Add dump meta command 

### PR DESCRIPTION
**Pull request by srascar**

### CODE REVIEW

File: Command/SonataDoctrineUtilsCommand.php [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+        return $allowedMeta;
+    }
+
+    /**
+     * @param \Doctrine\ORM\Mapping\ClassMetadata $meta
+     *
+     * @return array
+     */
+    private function normalizeMeta($meta)
```
Comment (rande) [inline Command/SonataDoctrineUtilsCommand.php]: Can you rename this to normalizeDoctrineORMMeta ?

File: Command/SonataDoctrineUtilsCommand.php [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+                'Get information on the current Doctrine\'s schema'
+            );
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->defaultDumpDirectory = $this->getContainer()->getParameter('sonata.doctrine.utils.dump_directory');
```
Comment (rande) [inline Command/SonataDoctrineUtilsCommand.php]: This should be overwritable with command line option

File: Command/SonataDoctrineUtilsCommand.php [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+        $output->writeln(sprintf('Initialising Doctrine metadata.'));
+        $manager = $this->getContainer()->get('doctrine')->getManager();
+        $metadata = $manager->getMetadataFactory()->getAllMetadata();
+
+        $allowedMeta = $this->filterMetadata($metadata, $input, $output);
+
+        /** @var \Doctrine\ORM\Mapping\ClassMetadata $meta */
+        foreach ($allowedMeta as $meta) {
+            $this->metadata[$meta->getName()] = $this->normalizeMeta($meta);
```
Comment (rande) [inline Command/SonataDoctrineUtilsCommand.php]: You should check if the meta is coming from doctrine ORM

File: Command/SonataDoctrineUtilsCommand.php [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->defaultDumpDirectory = $this->getContainer()->getParameter('sonata.doctrine.utils.dump_directory');
+
+        $output->writeln(sprintf('Initialising Doctrine metadata.'));
```
Comment (eko) [inline Command/SonataDoctrineUtilsCommand.php]: You could remove `sprintf()`, it is not necessary here

File: Command/SonataDoctrineUtilsCommand.php (hunk 2 above)
Comment (srascar) [inline Command/SonataDoctrineUtilsCommand.php]: It is already. It just prevent issues when using a wrong filename like "my_file.json"
(which would set directory to .)
but setting filename option to /my_directory/my_file.json works

**Comment by srascar** [comment]
@rande

**Comment by rande** [comment]
@srascar you need to rebase  and squash this PR ?

**Comment by rande** [comment]
@srascar you still need to rebase this PR, thanks

File: Command/SonataDoctrineUtilsCommand.php (hunk 2 above)
Comment (Bladrak) [inline Command/SonataDoctrineUtilsCommand.php]: You shouldn't need this parameter; default behavior should be to dump in current directory if no directory is specified.

File: DependencyInjection/SonataCoreExtension.php [hunk 5]
```diff
@@ -57,6 +57,7 @@ public function load(array $configs, ContainerBuilder $container)
         $loader->load('twig.xml');
         $loader->load('model_adapter.xml');
         $loader->load('core.xml');
+        $loader->load('utils.xml');
```
Comment (Bladrak) [inline DependencyInjection/SonataCoreExtension.php]: This is most of the time a bad idea to have a "utils" package/namespace/directory/...

File: Resources/config/utils.xml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <parameters>
+        <parameter key="sonata.doctrine.utils.dump_directory">%kernel.root_dir%/cache/sonata/ecommerce</parameter>
```
Comment (Bladrak) [inline Resources/config/utils.xml]: If you need a parameter, you most of the time rather need a configuration option or a function/command line parameter. Re-think this each time you face it.

File: Resources/doc/reference/command.rst [hunk 7]
```diff
@@ -0,0 +1,28 @@
+Command
+=======
+
+You can create a dump file with products schema information. Even those created with the previous command:
```
Comment (Bladrak) [inline Resources/doc/reference/command.rst]: ``You can dump products schema information into a file`` is more clear

Comment (Bladrak) [inline Resources/doc/reference/command.rst]: But I don't get why products? Does this concern only products? And what about the thing with the previous command?

File: Resources/doc/reference/command.rst [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+.. code-block:: bash
+
+    Usage:
+     sonata:doctrine:utils -r "/ProductBundle/" -f /tmp/dump.json
+
+    Options:
+     --filename (-f)       If filename is specified, result will be dump into this file under json format.
```
Comment (Bladrak) [inline Resources/doc/reference/command.rst]: ``dumped``

File: Command/SonataDoctrineUtilsCommand.php [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+     */
+    protected $metadata;
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('sonata:doctrine:utils')
```
Comment (Bladrak) [inline Command/SonataDoctrineUtilsCommand.php]: You should rename this command, it does not speak for what it does. Something like ``sonata:core:dump-doctrine-metadata`` is more self-explainable.

File: Resources/doc/reference/command.rst (hunk 8 above)
Comment (Bladrak) [inline Resources/doc/reference/command.rst]: -> If specified, outputs to given JSON file.

File: Resources/doc/reference/command.rst [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+     --entity-name (-E)    If entity-name is set, dump will only contain the specified entity and all it's extended classes.
```
Comment (Bladrak) [inline Resources/doc/reference/command.rst]: ``its``

File: Resources/doc/reference/command.rst [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+     --regex (-r)          If regex is set, dump will only contain entities which name match the pattern.
+     --help (-h)           Display this help message.
```
Comment (Bladrak) [inline Resources/doc/reference/command.rst]: ``Displays``

File: Resources/doc/reference/command.rst [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+     --ansi                Force ANSI output.
+     --no-ansi             Disable ANSI output.
+     --no-interaction (-n) Do not ask any interactive question.
+     --shell (-s)          Launch the shell.
+     --process-isolation   Launch commands from shell as a separate process.
+     --env (-e)            The Environment name. (default: "dev")
+     --no-debug            Switches off debug mode.
+
+
+This can be used for import/export purposes for example.
```
Comment (Bladrak) [inline Resources/doc/reference/command.rst]: ``for example`` is useless. Also, you need to be more verbose about what this command does.

**Comment by srascar** [comment]
done @rande

**Pull request merged**